File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(char c:s)
        {
            if(c==')')
            {
                if(!st.empty() && st.top()=='(')
                {
                    if(!st.empty())
                    st.pop();
                }
                else return false;
            }
            else if(c=='}')
            {
                if(!st.empty() && st.top()=='{')
                {
                    st.pop();
                }
                else return false;
            }
            else if(c==']')
            {
                if(!st.empty() && st.top()=='[')
                {
                    st.pop();
                }
                else return false;
            }
            else 
                st.push(c);
        }
        if(!st.empty())
            return false;
        return true;
    }
};

File: 20-valid-parentheses/20-valid-parentheses.go
func isValid(s string) bool {
    stack := make([]rune, 0, len(s))
    for _, c := range s {
        switch c {
        case '(':
            stack = append(stack, ')')
        case '{':
            stack = append(stack, '}')
        case '[':
            stack = append(stack, ']')
        default:
            if len(stack) == 0 || c != stack[len(stack)-1] {
                return false
            }
            stack = stack[:len(stack)-1]
        }
    }
    return len(stack) == 0
}